app/models/tag.rb

```diff
@@ -52,6 +52,7 @@ class Tag < ActsAsTaggableOn::Tag
   validate :validate_name, if: :name?
 
   before_validation :evaluate_markdown
+  before_validation :tidy_short_summary
```

Comment:
There are two possible strategies here for problematic input

- inform the user we won't accept their input (user facing validation error), prompt them to resubmit
- change the user input to a safe format before persisting/presenting (especially when it would otherwise be valid)

The second strategy could have negative interactions if we then also validated the modified value (if we had another rule like "validate short_summary present" and the tidying left it empty - we would show a confusing "short summary can't be empty" message after they provided non empty input (consisting only of tags).

We're not doing that here so it's not a big potential for surprise, and Tags aren't open to mass modification (typically moderators and admins), but it seems like in many cases Forem operates using the first strategy (reject and explain) rather than the second (correct and continue).

Since Tags don't change much, and this check is cheap and comparatively non-obtrusive, the pre-validation check makes sense here.

Reply:
This is great context on considering approaches.